scripts/tests/sanitycheck/test_testsuite_class.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+# pylint: disable=line-too-long
+# pylint: disable=C0321
+'''
+This test file contains testcases for Testsuite class of sanitycheck
+'''
+import sys
+import os
+
+ZEPHYR_BASE = os.getenv("ZEPHYR_BASE")
+sys.path.insert(0, os.path.join(ZEPHYR_BASE, "scripts/sanity_chk"))
+
+from sanitylib import TestCase
+
+def test_testsuite_add_testcases(class_testsuite):
+    """ Testing add_testcase function of Testsuite class in sanitycheck """
+    # Test 1: Check the list of testcases after calling add testcases function is as expected
+    class_testsuite.SAMPLE_FILENAME = 'test_sample_app.yaml'
+    class_testsuite.TESTCASE_FILENAME = 'test_data.yaml'
+    class_testsuite.add_testcases()
+    expected_testcases = ['scripts/tests/sanitycheck/test_data/testcases/tests/test_b/kernel.fifo.timeout',
+                          'scripts/tests/sanitycheck/test_data/testcases/tests/test_b/kernel.fifo.timeout.poll',
+                          'scripts/tests/sanitycheck/test_data/testcases/tests/test_c/kernel.fifo.usage',
+                          'scripts/tests/sanitycheck/test_data/testcases/tests/test_c/kernel.fifo.usage.poll',
+                          'scripts/tests/sanitycheck/test_data/testcases/tests/test_a/kernel.fifo',
+                          'scripts/tests/sanitycheck/test_data/testcases/tests/test_a/kernel.fifo.poll',
+                          'scripts/tests/sanitycheck/test_data/testcases/samples/test_app/sample.app_dev.external_lib']
+    assert set(class_testsuite.testcases.keys()) == set(expected_testcases)
+
+    # Test 2 : Assert Testcase name is expected & all the testcases values are testcase class objects
+    testcase = class_testsuite.testcases.get('scripts/tests/sanitycheck/test_data/testcases/tests/test_a/kernel.fifo')
+    assert testcase.name == 'scripts/tests/sanitycheck/test_data/testcases/tests/test_a/kernel.fifo'
+    assert all(isinstance(n, TestCase) for n in class_testsuite.testcases.values())
+
+    # Test 3: Assert testcase object has all the expected attributes
+    list_of_vars = []
+    for _, test_case in class_testsuite.testcases.items():
+        for i in test_case.__dict__.items():
+            testcase_vars = list(i)
+            list_of_vars.append(testcase_vars[0])
+        expected_variables_to_check = ['id', 'source_dir', 'yamlfile', 'cases', 'name',
+                                       'extra_args', 'type', 'tags', 'extra_configs', 'arch_whitelist', 'arch_exclude',
+                                       'skip', 'platform_exclude', 'platform_whitelist', 'toolchain_exclude', 'toolchain_whitelist',
+                                       'tc_filter', 'timeout', 'harness', 'harness_config', 'build_only', 'build_on_all', 'slow',
+                                       'min_ram', 'depends_on', 'min_flash', 'extra_sections']
+        assert set(expected_variables_to_check) == set(list_of_vars)
```

Comment:
Not how is this a test, I would rather see those tested when used, and not just list all instances variables and see if they are there.

Reply:
Removed this test scenario.